src/Exception/InvalidRequestException.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GuzzleHttp\Exception;
+
+use Psr\Http\Message\RequestInterface;
+
+class InvalidRequestException extends \InvalidArgumentException implements \Psr\Http\Client\Exception\RequestException,  GuzzleException
```

Comment:
Maybe you could have a look on this https://github.com/guzzle/guzzle/pull/2163. If we could merge this PR you can have this exception extend that one.